### DISCUSSION TITLE
Node ID Uniqueness

### DISCUSSION THREAD

**Discussion by flu0r1ne** [tree-sitter/tree-sitter#3549] [Q&A] [answered]
If one or more trees are parsed de novo within the same program (e.g. not incrementally parsing the same document), are the node IDs guaranteed to be unique? Specifically, are there no two identical IDs within the entire set of active IDs across all trees?

It seems that the ID field is a generic pointer, implying that uniqueness should be guaranteed, as it is unlikely that the same node will be allocated for two separate documents. However, I am uncertain if this behavior should be considered an API guarantee.

**Comment by ObserverOfTime** [comment, marked as answer]
Yes, node IDs are unique pointers.
